lang/cs/Org.Apache.REEF.Network/Elastic/Task/IIdentifiable.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using Org.Apache.REEF.Utilities.Attributes;
+
+namespace Org.Apache.REEF.Network.Elastic.Task
+{
+    /// <summary>
+    /// Interface used to return an identfiable object from a class.
+    /// </summary>
+    [Unstable("0.16", "API may change")]
+    public interface IIdentifiable
+    {
+        /// <summary>
+        /// The identifier for the node.
+        /// </summary>
+        NodeIdentifier NodeId { get; }
```

Comment:
I have a better idea: how about we delete this interface altogether, and instead add an extension method - say, at the end of the `NodeIdentifier.cs` class we just add:
```C#
public static class NodeIdentifierUtils
{
    public static NodeIdentifier NodeId(this NodeIdentifier id)
    {
        return id;
    }

    public static NodeIdentifier NodeId(this INodeIdentifier id)
    {
        return new NodeIdentifier(id.StageName, id.OperatorId);
    }
}
```
then we don't have to implement and add the `.NodeId` property everywhere - instead, any class that implements `INodeIdentifier` interface will automagically have `.NodeId()` method that returns exactly what we need

Reply:
Fixed!